docs/end-user/stack-samples-and-call-trees.md

```diff
@@ -0,0 +1,55 @@
+
+The call node with the function `B` continues to have the running time of 5ms, but also has a self time of 2ms. Function `C` has a reduced running time of `3ms` and finally do work was only observed `3ms` for both the running and self time.
+
+In summary, self time is where work was actually happening when observed by the profiler.
+
+## How to form a call tree
+
+> Note: Feel free to follow along with a real call tree that reproduces this structure: [https://perfht.ml/2w45IdC](https://perfht.ml/2w45IdC)
+
+The simple call tree above did not take into account any branching stacks. It only concerned itself with running and self time. This section will dive into how the call tree handles stacks of different shapes.
+
+![This image shows contiguous stacks, and how they are sampled over time. It fades out as it goes left to right to indicate that samples are gathered for long periods of time. The stacks are labeled by 1 millisecond intervals. The stacks are labeled by functions that are single capital letters, ranged A to H.](../images/samples.svg)
+
+Consider the first 3 samples in this profile.
+
+![On the left of this graph are three samples with different stacks. A, B, C, D, E. A, B, C, F, G. A, B, H, F, G. The bottom row with A is all grouped together. The second row with B is all grouped together. The third row has the two C functions grouped together, and the H distinct. All the rest of the function sare not grouped. The right graphic demonstrates the shape of the call tree.](../images/call-tree.svg)
+
+These three samples share things in common. The root of each stack is the function `A`. These could be aggregated together into a single node. The same happens for node `B`. However the next level with the function `C` and `H` differ. At this point the call tree splits into two different nodes. There is a node for the function `C`, and the function `H`.
+
+The call tree will continue merging functions that are at the same level, but only if they are part of the current branching pattern. For instance, consider function `F` and `G`. While both functions are at the same level of the stack, they are at different branching points. The middle `F -> G` stacks have `C` as their ancestor, and the right `F -> G` stacks have `H` as their ancestor, and thus are not grouped together.
+
+A useful exercise at this point would be to consider [the running time](../images/call-tree-running-time.svg) and [self time](../images/call-tree-self-time.svg) of each node this call tree.
+
+# Referring to call nodes in the call tree
+
+The above example shows how functions F and G can appear multiple times in a call tree, thus it's ambiguous to refer to a call node by only its function name. Call nodes can be referred to by their call node paths, which are composed of a list of functions from root to leaf. The middle `G` call node in the tree would have the call node path `A, B, C, F, G`, while the one on the right would have the call node path `A, B, H, F, G`.
```

Comment:
I've mixed feelings about the fact that this paragraph should be part of the end-user docs or not. Let's keep it for now but I'm curious to hear your opinion.